lib/src/data/response/newest_article_response.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:allthenews/src/data/response/multimedia.dart';
+import 'package:allthenews/src/domain/model/article.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'newest_article_response.g.dart';
+
+@JsonSerializable()
+class NewestArticleResponse {
+  final int id;
+
+  final String url;
+
+  @JsonKey(name: 'updated_date', fromJson: DateTime.parse)
+  final DateTime updated;
+
+  @JsonKey(name: 'byline')
+  final String author;
+
+  final String title;
+
+  final String abstract;
+
+  final List<Multimedia> multimedia;
+
+  const NewestArticleResponse(
+    this.id,
+    this.url,
+    this.updated,
+    this.author,
+    this.title,
+    this.abstract,
+    this.multimedia,
+  );
+
+  static NewestArticleResponse fromJson(Map<String, dynamic> json) =>
+      _$NewestArticleResponseFromJson(json);
+
+  static Article toArticle(NewestArticleResponse response) => Article(
+        id: response.id,
+        authorName: response.author,
+        title: response.title,
+        abstract: response.abstract,
+        date: "${response.updated.year}-${response.updated.month}-${response.updated.day}",
```

Comment:
w sumie jakby nie patrzeć jest to mapowanie stricte pod ui, jakby dane z backendu przychodziły już tak jak mają zostać wyświetlone na ui to date i time jako string spoko, ale w przypadku jakichś mapowań z responsa na domenę bym został przy DateTime, a jak gdzieś spływa nawet jako string, to mimo wszystko mapował na DateTime w domenie i potem napisał jakiegoś DateTimeFormattera który zajmowałby się formatowaniem pod ui :)